### PR TITLE
storage: also move for inmem IngestExternalFile

### PR DESCRIPTION
**Pull request by danhhz**
This was originally special cased for in-memory rocksdb instances to
work around a bug in rocksdb, which has since been fixed. Remove the
special case.

### CODE REVIEW

File: pkg/storage/replica_proposal.go [hunk 1]
```diff
@@ -581,15 +581,12 @@ func addSSTablePreApply(
...
 		path = fmt.Sprintf("%x", checksum)
-		move = false
 		if err := inmem.WriteFile(path, sst.Data); err != nil {
 			panic(err)
 		}
 	} else {
-		move = true
 		// TODO(tschottdorf): remove this once sideloaded storage guarantees its
 		// existence.
 		if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil {
```
Comment (tschottdorf) [inline pkg/storage/replica_proposal.go]: BTW, you should be able to remove this, too.

**Review by tbg — APPROVED** [review]

File: pkg/storage/replica_proposal.go (hunk 1 above)
Comment (danhhz) [inline pkg/storage/replica_proposal.go]: 👍 will do in another PR

**Comment by danhhz** [comment]
TFTR!
